cpp/src/util/memory_pool.cpp

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <raft/util/memory_pool-inl.hpp>
```

Comment:
It's nice to see this centralized and broken up but I also want to make the note that we should never be creating memory pools outside of the user's control. It's been on the todo list to remove from the codebase the instances where we are creating memory pools on the user's behalf. The user should always be creating those.

Reply:
We don't need to make this change in this PR btw- just mentioning the goal.